File: app/src/main/java/com/martynaskairys/walltip/utils/ActivityUtils.java
package com.martynaskairys.walltip.utils;

public class ActivityUtils {
    @SafeVarargs
    public static void startActivityWithTransitionAnimation(Activity activity, Intent intent, Pair<View, String>... sharedViews) {
        ActivityOptionsCompat ao = ActivityOptionsCompat.makeSceneTransitionAnimation(activity, sharedViews);
        Bundle options = ao.toBundle();
        activity.startActivity(intent, options);
    }
}

File: app/src/main/java/com/martynaskairys/walltip/shared/utils/ActivityUtils.kt
package com.martynaskairys.walltip.shared.utils

import android.app.Activity
import android.content.Intent
import android.support.v4.app.ActivityOptionsCompat
import android.support.v4.util.Pair
import android.view.View

object ActivityUtils {

    @SafeVarargs
    fun startActivityWithTransitionAnimation(activity: Activity, intent: Intent, vararg sharedViews: Pair<View, String>) {
        val ao = ActivityOptionsCompat.makeSceneTransitionAnimation(activity, *sharedViews)
        val options = ao.toBundle()
        activity.startActivity(intent, options)
    }
}